### DISCUSSION TITLE
android ndk not support c++20 modules ?

### DISCUSSION THREAD

**Discussion by kaizer1** [android/ndk#1795] [General]
An example for everyone
  I'm trying to connect modules to my ndk project and nothing happens. Doing in Gragle cppFlags:                cppFlags "-std=c++20 -Ofast -O3 -ftree-vectorize -fmodules -fmodule-map-file=Q:/AndroidSDK/ndk/25.0.8775105/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/module.modulemap" 

And get this error: 
-funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -std=c++20 -Ofast -O3 -ftree-vectorize -fmodules -fmodule-map-file=Q:/AndroidSDK/ndk/25.0.8775105/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/module.modulemap -fno-limit-debug-info  -fPIC -fopenmp -MD -MT CMakeFiles/renderingsLos.dir/src/main/cpp/renderingStart.cpp.o -MF CMakeFiles\renderingsLos.dir\src\main\cpp\renderingStart.cpp.o.d -o CMakeFiles/renderingsLos.dir/src/main/cpp/renderingStart.cpp.o -c Q:/AndroidProject/Renderings/app/src/main/cpp/renderingStart.cpp
  While building module 'std' imported from Q:/AndroidSDK/ndk/25.0.8775105/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/stdint.h:123:
  In file included from <module-includes>:14:
  In file included from Q:/AndroidSDK/ndk/25.0.8775105/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1/locale.h:42:
  In file included from Q:/AndroidSDK/ndk/25.0.8775105/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/local/include/locale.h:31:
  Q:/AndroidSDK/ndk/25.0.8775105/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/locale.h:103:1: error: missing '#include <xlocale.h>'; 'locale_t' must be declared before it is used
  locale_t duplocale(locale_t __l) __INTRODUCED_IN(21);

Basically, this is from connecting -fmodules right after this such errors (xlocale main error) and so on - two more in fact. But I can't figure out how to bypass them. In general, support for c++ modules is implemented in NDK, and if so, how to connect them? Thank you for your attention

System Windows 11, ndk version '25.0.8775105'
